### PR TITLE
Initial CI deployment script

### PR DESCRIPTION
**Pull request by szkl**
Signed-off-by: Sonmez Kartal <sonmez@koding.com>

### CODE REVIEW

File: scripts/ci/deploy [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		declare archive_name=$timestamp-$revision
+		declare version_label=$archive_name-$branch-$environment
+		declare archive_file=$version_label.zip
+
+		./configure \
+			--without-envvars \
+			--config $config \
+			--ebEnvName $environment
+
+		source .env.sh
```
Comment (cihangir) [inline scripts/ci/deploy]: do you need this line? we already source at line 31

File: scripts/ci/deploy [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+		declare bucket=${!env_s3_bucket:-$KONFIG_CI_AWS_S3_BUCKET}
+		bucket=$(echo $bucket | env --ignore-environment region=$region envsubst)
+
+		aws s3 cp --acl private $archive_file s3://$bucket/
+
+		aws elasticbeanstalk create-application-version \
+			--application-name koding \
+			--version-label $version_label \
+			--source-bundle $(jo S3Bucket=$bucket S3Key=$archive_file)
```
Comment (cihangir) [inline scripts/ci/deploy]: what is `jo`?

File: scripts/ci/deploy [hunk 3]
```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
```
Comment (cihangir) [inline scripts/ci/deploy]: could you make sure this works on macOS too?

**Review by cihangir — COMMENTED** [review]
LGTM, except, i have no idea about `jo`.

File: scripts/ci/deploy (hunk 2 above)
Comment (rjeczalik) [inline scripts/ci/deploy]: Isn't it [this](http://jpmens.net/2016/03/05/a-shell-command-to-create-json-jo/)? So basically `jo S3Bucket=$bucket S3Key=$archive_file` produces:

```json
$ jo S3Bucket=$bucket S3Key=$archive_file
{"S3Bucket": "$bucket", "S3Key": "$archive_file"}
```

Comment (cihangir) [inline scripts/ci/deploy]: yep, most probably it is that one. but why i commented like that is, i don't remember merging anything to infra repo with "jo" installation/executable ( playing the [dumb](https://github.com/koding/eb-deploy/blob/master/run.sh#L68) here :) )

Comment (szkl) [inline scripts/ci/deploy]: Circle CI is running [koding/circle](https://github.com/koding/docker-circle/blob/master/Dockerfile) image which will be added into koding/infra with other Dockerfiles very soon.

File: scripts/ci/deploy (hunk 1 above)
Comment (szkl) [inline scripts/ci/deploy]: This is a must to read credentials for environment specific credentials. L31 is probably not necessary, though. Good point.

File: scripts/ci/deploy (hunk 3 above)
Comment (szkl) [inline scripts/ci/deploy]: 👍

Comment (szkl) [inline scripts/ci/deploy]: I've made changes for macOS.

File: scripts/ci/deploy (hunk 1 above)
Comment (cihangir) [inline scripts/ci/deploy]: 👍

File: scripts/ci/deploy (hunk 3 above)
Comment (cihangir) [inline scripts/ci/deploy]: 🎉

**Review by cihangir — APPROVED** [review]

**Review by cihangir — APPROVED** [review]

**Pull request merged**